### PR TITLE
feat(docker): create ready-to-install docker deployment

### PR DESCRIPTION
**Pull request by tcrasset**

### CODE REVIEW

File: docker/Makefile [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+.PHONY: secrets-with-email-auth
+
+
+$(SECRETS_DIR)/db_password:
+	mkdir -p $(SECRETS_DIR)
+	python3 -c "import secrets; print(secrets.token_hex())" > $@
+
+$(SECRETS_DIR)/db_user:
+	mkdir -p $(SECRETS_DIR)
+	touch $@
```
Comment (tcrasset) [inline docker/Makefile:28]: Note that I didn't want to have a default value here, so I just used `touch` and not `echo ""` which might have created a database user with an empty name (not sure)

File: docker/.env.sample [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Copy this file and name it `.env.`
+# See https://docs.octopize.io/docs/deploying/self-hosted/configuration
+
+ENV_NAME="company-prod"
+DB_HOST="db"
+
+# When serving locally (check entries in /etc/hosts):
+BASE_API_URL='http://api.company.local:8000'
+WEB_CLIENT_URL='http://www.company.local:3000'
```
Comment (tas17) [inline docker/.env.sample:9]: why `company`?

File: docker/Makefile [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+# File-based recipes
+
+common-secrets: $(SECRETS_DIR)/db_password $(SECRETS_DIR)/db_name $(SECRETS_DIR)/db_user $(SECRETS_DIR)/pepper $(SECRETS_DIR)/file_encryption_key $(SECRETS_DIR)/authjwt_secret_key $(SECRETS_DIR)/organization_name
+.PHONY: common-secrets
+
+secrets-with-username-auth: common-secrets $(SECRETS_DIR)/avatar_first_user_password $(SECRETS_DIR)/avatar_first_user_name ## Initialize secrets for username authentication
+.PHONY: secrets-with-username-auth
+
+secrets-with-email-auth: common-secrets $(SECRETS_DIR)/admin_emails $(SECRETS_DIR)/aws_mail_account_access_key_id $(SECRETS_DIR)/aws_mail_account_secret_access_key ## Initialize secrets for email authentication
+.PHONY: secrets-with-email-auth
```
Comment (tas17) [inline docker/Makefile:19]: Nice!

**Review by tas17 — APPROVED** [review]
Nice! I just find this `company` thing not clear but that's nit

File: docker/.env.sample (hunk 2 above)
Comment (tcrasset) [reply docker/.env.sample:9]: This is for our potential clients to use too, which might not use our api, so I didn't want to put `api.octopize.local`, I wanted it to be generic.

Any suggestions

**Pull request merged**